test/python/transpiler/test_check_cnot_direction.py

```diff
@@ -0,0 +1,317 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Test the Check CNOT direction pass"""
+
+import unittest
+
+from qiskit import QuantumRegister, QuantumCircuit
+from qiskit.transpiler.passes import CheckCnotDirection
+from qiskit.mapper import CouplingMap
+from qiskit.converters import circuit_to_dag
+from qiskit.test import QiskitTestCase
+
+
+class TestCheckCNotDirection(QiskitTestCase):
+    """ Tests the CheckCnotDirection pass with CX gates"""
+
+    def test_trivial_map(self):
+        """ Trivial map in a circuit without entanglement
+         qr0:---[H]---
+
+         qr1:---[H]---
+
+         qr2:---[H]---
+
+         CouplingMap map: None
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.h(qr)
+        coupling = CouplingMap()
+        dag = circuit_to_dag(circuit)
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+        self.assertTrue(pass_.property_set['is_direction_mapped'])
+
+    def test_true_direction(self):
+        """ Mapped is easy to check
+         qr0:--(+)-[H]-(+)-
+                |       |
+         qr1:---.-------|--
+                        |
+         qr2:-----------.--
+
```

Comment:
I think the `(+)`s and `.`s are swapped in the diagram.

Reply:
fixed in f383590d